modules/oe_migration_workflow/src/Plugin/migrate/process/SetWorkflowState.php

```diff
@@ -0,0 +1,243 @@
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    // The default configuration.
+    $this->configuration += [
+      'published_state' => 'published',
+      'unpublished_state' => 'draft',
+      'status_field_name' => 'status',
+    ];
+    $this->entityTypeManager = $entity_type_manager;
+    // Check if the configuration is set correctly.
+    $this->validateConfigurationKeys();
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\migrate\MigrateException
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition, MigrationInterface $migration = NULL) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+    // The expected value has to be a string but the source entity might not
+    // have a moderation state defined, in that case stringify the NULL value
+    // and continue.
+    $value = is_null($value) ? (string) $value : $value;
+    // Check if the source value is of the expected type.
+    if (!is_string($value)) {
+      throw new MigrateException(sprintf('%s is not a string.', var_export($value, TRUE)));
+    }
+    // Check whether the current workflow state is a valid one or not. If it's
+    // not a valid one, set it according to the entity "status". In this case,
+    // if the status is set to "1", set it to the given "published_state", in
+    // any other case, set it to the given "unpublished_state".
+    $workflow = $this->getWorkflow($this->configuration['workflow_config_name']);
+    $status_field = $this->configuration['status_field_name'];
+    if ($this->isValidWorkflowState($workflow, $value) === FALSE) {
+      $value = (int) $row->getSourceProperty($status_field) === 1 ? $this->configuration['published_state'] : $this->configuration['unpublished_state'];
+    }
+    // Check whether the entity status and the workflow state status match. If
+    // it's not the case, throw a migration exception, otherwise proceed with
+    // the operation.
+    if ((int) $row->getSourceProperty($status_field) !== (int) $workflow->get('type_settings')['states'][$value]['published']) {
+      throw new MigrateException('The entity status and the workflow state status don\'t match.');
+    }
+
+    return $value;
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
```

Comment:
All @throws must be removed.

Reply:
Done, I misunderstood the first comment about it. I already removed all @thows .